plugins/node/opentelemetry-instrumentation-aws-sdk/src/propwrap.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This block is derived from esbuild's bundling support.
+ * https://github.com/evanw/esbuild/blob/v0.14.42/internal/runtime/runtime.go#L22
+ *
+ * License:
+ * MIT License
+ *
+ * Copyright (c) 2020 Evan Wallace
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+const __defProp = Object.defineProperty;
+const __getOwnPropDesc = Object.getOwnPropertyDescriptor;
+const __hasOwnProp = Object.prototype.hasOwnProperty;
+const __getOwnPropNames = Object.getOwnPropertyNames;
+const __copyProps = (
+  to: any,
+  from: any,
+  except: string,
+  desc?: PropertyDescriptor | undefined
+) => {
+  if ((from && typeof from === 'object') || typeof from === 'function') {
+    for (const key of __getOwnPropNames(from)) {
+      if (!__hasOwnProp.call(to, key) && key !== except) {
+        __defProp(to, key, {
+          get: () => from[key] as any,
+          enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable,
+        });
+      }
+    }
+  }
+  return to;
+};
+
+/**
+ * Derived from https://github.com/elastic/apm-agent-nodejs/blob/main/lib/propwrap.js
```

Comment:
I am sure donating it will be fine... but yah... I'll get my team of lawyers to talk to your team of lawyers... <weak smile/>

Do you know if there is any prior art of including license text from other adapted pieces of code in the core or contrib repos, so I can crib?

Reply:
> I am sure donating it will be fine

I'll ask internally at Elastic and follow-up.

Reply:
Donating this function is fine. Done in commit 0fcca666.

Reply:
Thanks for taking the time to sort this out :slightly_smiling_face: